If we don't use --with-sysroot for cross-compilers, the $LIBPATH_SUFFIX 
setting is ignored when generating the default lib search path.

http://sources.redhat.com/bugzilla/show_bug.cgi?id=2755

2006-09-24  Mike Frysinger  <vapier@gentoo.org>

	* genscripts.sh: Respect LIBPATH_SUFFIX when not using sysroot.

--- binutils/ld/genscripts.sh
+++ binutils/ld/genscripts.sh
@@ -186,6 +186,14 @@
   ::) LIB_PATH=${tool_lib} ;;
   *) LIB_PATH=${tool_lib}:${LIB_PATH} ;;
   esac
+  # For multilib targets, search both $tool_lib dirs
+  if [ "x${LIBPATH_SUFFIX}" != "x" ] ; then
+    case :${LIB_PATH}: in
+      ::: | *:${tool_lib}${LIBPATH_SUFFIX}:*) ;;
+      ::) LIB_PATH=${tool_lib}${LIBPATH_SUFFIX} ;;
+      *) LIB_PATH=${tool_lib}${LIBPATH_SUFFIX}:${LIB_PATH} ;;
+    esac
+  fi
 fi
 
 LIB_SEARCH_DIRS=`echo ${LIB_PATH} | sed -e 's/:/ /g' -e 's/\([^ ][^ ]*\)/SEARCH_DIR(\\"\1\\");/g'`
